libgraphics/source/AssetsStorage/AssetStorage_build.cxx

```diff
@@ -0,0 +1,201 @@
+#include "pivot/graphics/AssetStorage.hxx"
+
+#include "pivot/graphics/DebugMacros.hxx"
+#include "pivot/graphics/vk_debug.hxx"
+
+namespace pivot::graphics
+{
+
+void AssetStorage::build(DescriptorBuilder builder, BuildFlags flags)
+{
+    DEBUG_FUNCTION
+    // check for incorrect combination of flags
+    assert(std::popcount(static_cast<std::underlying_type_t<AssetStorage::BuildFlagBits>>(flags)) == 1);
```

Comment:
Why do you use a flag type if only one can be active at a time ? Just use an enum.

Reply:
It is just temporary. More options will appear in future PR

Reply:
If they are exclusive, store them on the same two bits field, instead of having one bit per exclusive option.